### PR TITLE
Fix implicit optional

### LINKED ISSUE #3918: Fix mypy issues
## Expected behavior ##

The latest release of mypy seems to be identify new issues, see: https://github.com/MDAnalysis/mdanalysis/actions/runs/3432903207/jobs/5722653305

These are making CI fail, we should address them.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #3918 

PR Checklist
------------
 - Tests?
 - Docs?
 - CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
Huh, looks like I forgot to turn off cron CI from being activated on PRs, doing so here.

**Comment by IAlibay** [comment]
Seems like the only thing that was needed, let's get this merged so we can CI back to green.

File: .github/workflows/gh-ci-cron.yaml [hunk 1]
```diff
@@ -3,9 +3,6 @@ on:
   schedule:
     # 3 am Tuesdays and Fridays
     - cron: "0 3 * * 2,5"
-  pull_request:
```
Comment (hmacdope) [inline .github/workflows/gh-ci-cron.yaml:6]: Do we still need this specificity to develop ?

**Review by hmacdope — COMMENTED** [review]
~~Quick question on chron job :)~~ 

Ignore me

File: .github/workflows/gh-ci-cron.yaml (hunk 1 above)
Comment (hmacdope) [reply .github/workflows/gh-ci-cron.yaml:6]: Ignore me my phone wasn't showing your previous comment @IAlibay

**Review by hmacdope — APPROVED** [review]
LGTM

**Review by orbeckst — APPROVED** [review]
ok

**Pull request merged**